packages/generator/src/operations/return-type.ts

```diff
@@ -31,11 +36,24 @@ export function operationReturnType({
   if (returnType.isNullable) {
     type = `${type} | null`;
   }
-  type = wrapRequestBuilderAroundType(
-    requestBuilderName,
-    parametersTypeName,
-    type
-  );
+
+  if (isBound && !boundEntitySetName) {
+    throw new Error(
```

Comment:
[q]/[req] I assume this will stop the generation?
For people who's using our generator and their specs do not meet our expectation, this will be a breaking change.
Before, the bound functions are ignored, and now the generation will be broken.
We should ignore this case to avoid the breaking changes.

Reply:
I'm not sure if I agree here. Yes this is breaking for invalid edmx files, but is our policy to always avoid this? I think that this could be handled with a compatibility note and allow us to ship better software if we clearly define what we expect, rather than having more warnings.

Reply:
As a general generator CLI design (which is done only in the openapi generator), we provide an option `--skipValidation`, so the user can still finish the generation and use the generated client.
In the long term, the odata generator should also have this option ([ticket](https://github.com/SAP/cloud-sdk-backlog/issues/506)) and it should not block the users.

Currently, the policy is, we allows limited breaking changes for bug fix or improvement and mark them in the compatibility notes.

Therefore, both of the following options are valid:
1. show warning instead of error and wait for v3
2. show error and stop the generation and mark it in the compatibility notes + support potential issues

As we are going to work on the generator [milestone](https://github.com/SAP/cloud-sdk-backlog/milestone/19) soon, I would prefer 1, because we will then have the `--skipValidation` option, and the major version bump better fits breaking changes.

Reply:
@FrankEssenberger , please leave your opinion about this.

Reply:
I think this case is just a indicator that we messed something up. If the method isBound there must be an entityClassName since it is bound. So this should never happen at runtime. However, since the property is optional for the unbound case we need a check.

Reply:
Discussed with Frank.
This check turns out to be related to the ts implementation, as `boundEntitySetName` is optional.
An invalid spec will not trigger this error.